compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4882,12 +4887,24 @@ public BType getSafeType(BType type, boolean liftNil, boolean liftError) {
         }
 
         if (liftError) {
-            errorLiftedType.remove(symTable.errorType);
+            LinkedHashSet<BType> bTypes = new LinkedHashSet<>();
+            for (BType t : errorLiftedType.getMemberTypes()) {
+                if (t.tag != TypeTags.ERROR) {
+                    bTypes.add(t);
+                }
+            }
+            memTypes = bTypes;
+            errorLiftedType = BUnionType.create(null, memTypes);
         }
 
         if (errorLiftedType.getMemberTypes().size() == 1) {
             return errorLiftedType.getMemberTypes().toArray(new BType[0])[0];
         }
+
+        if (errorLiftedType.getMemberTypes().size() == 0) {
+            return symTable.neverType;
```

Comment:
For now, shall we return semanticError? I don't think we've properly introduced type checking with never yet.

Reply:
sure